tmt/utils.py

```diff
@@ -1279,6 +1289,44 @@ def copytree(
     return dst
 
 
+def get_full_metadata(fmf_tree_path: str, node_path: str) -> Any:
+    """
+    Get full metadata for a node in any fmf tree
+    Go through fmf tree nodes using given relative node path
+    and return full data as dictionary
+    """
+    return fmf.Tree(fmf_tree_path).find(node_path).data
+
+
+def filter_paths(
+        directory: Path, searching: List[str],
+        files_only: bool = False) -> List[Optional[Path]]:
```

Comment:
Why `Optional[Path]`? Will the list be `[Path1, None, None, Path2, ...]`?

Reply:
Most likely not None in between other items, that would be weird, but it can find no files, then it would be an empty list. That's why Optional is here.

Reply:
Then you don't need `Optional`, even an empty list is `List[Path]`. The position of `Optional` means *items* of the list might be `None`, not the return value itself, which would spell `Optional[List[Path]]` - and I'd advise against doing so, an empty list would be much better than returning `None`. `[]` still carries "no paths found" semantics, it's false-ish when tested, can be iterated over safely, with no extra preliminary test (`if filtered_paths is not None: for path in filtered_paths: ...`) - `for path in <an empty list>` would have the same effect with less noise.

Reply:
allright, didn't know it works that way, changed to List[Path]